### ISSUE TITLE
Grep filter do not work in 1.1.0 (master)

### ISSUE THREAD

**Issue by nikinaydenov** [fluent/fluent-bit#993]
## Bug Report

**Describe the bug**
Grep filter do not work on binary compiled from master: d38c1df2d617da4f81ad82cdf5512f5537c7c494

**To Reproduce**
- Rubular link if applicable:
- Example log message if applicable:
```
echo '{"key1": 123456789, "message": "info here", "to_remove": "this", "log": "grep this: please"}' | /opt/td-agent/embedded/bin/fluent-cat test_tag
```
- Steps to reproduce the problem:
Use this config:
```
[SERVICE]
    Flush        5
    Daemon       Off
    Log_Level    trace
    Parsers_File parsers.conf
    #Plugins_File plugins.conf
    HTTP_Server  Off

[INPUT]
    Name        forward
    Listen      127.0.0.1
    Port        24224
    Chunk_Size  32  
    Buffer_Size 64  

[FILTER]
    Name    grep
    Match   test_tag
    Exclude log grep this:

[OUTPUT]
    Name  stdout
    Match *
```

**Expected behavior**
The line should be filtered (not otputed)

**Screenshots**
```
Fluent Bit v1.1.0
Copyright (C) Treasure Data

[2018/12/20 15:18:31] [debug] [storage] [cio stream] new stream registered: forward.0
[2018/12/20 15:18:31] [ info] [storage] initializing...
[2018/12/20 15:18:31] [ info] [storage] in-memory
[2018/12/20 15:18:31] [ info] [storage] normal synchronization mode, checksum disabled
[2018/12/20 15:18:31] [ info] [engine] started (pid=29669)
[2018/12/20 15:18:31] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2018/12/20 15:18:31] [debug] [in_fw] Listen='127.0.0.1' TCP_Port=24224
[2018/12/20 15:18:31] [ info] [in_fw] binding 127.0.0.1:24224
[2018/12/20 15:18:31] [debug] [router] match rule forward.0:stdout.0
[0] test_tag: [1545311915.710455400, {"key1"=>123456789, "message"=>"info here", "to_remove"=>"this", "log"=>"grep this: please"}]
[2018/12/20 15:18:35] [debug] [task] created task=0x5578eaef3de0 id=0 OK
[2018/12/20 15:18:35] [debug] [task] destroy task=0x5578eaef3de0 (task_id=0)
[0] test_tag: [1545311916.242525260, {"key1"=>123456789, "message"=>"info here", "to_remove"=>"this", "log"=>"grep this: please"}]
[1] test_tag: [1545311916.801286694, {"key1"=>123456789, "message"=>"info here", "to_remove"=>"this", "log"=>"grep this: please"}]
[2018/12/20 15:18:40] [debug] [task] created task=0x5578eaee7370 id=0 OK
[2018/12/20 15:18:40] [debug] [task] destroy task=0x5578eaee7370 (task_id=0)
```
**Your Environment**
* Version used: 1.1.0 build from master 
* Configuration: see above
* Environment name and version (e.g. Kubernetes? What version?): 
* Server type and version: laptop
* Operating System and version: Ubuntu 18.0.4
* Filters and plugins: Grep alone 

**Additional context**
The same compiled from git checkout v0.14.9 works fine. v1.0.0 do not build clean, so can not test it

**Comment by nokute78** [comment]
@edsiper 

flb_filter tries to reset content length with `flb_input_chunk_write_at`.
```c
            /* Override buffer just if it was modified */
            if (ret == FLB_FILTER_MODIFIED) {
                /* all records removed, no data to continue processing */
                if (out_size == 0) {
                    /* reset data content length */
                    flb_input_chunk_write_at(ic, content_size, "", 0);
```

However cio ignores it because data size is zero.
```c
int cio_memfs_write(struct cio_chunk *ch, const void *buf, size_t count)
{
    size_t av_size;
    size_t new_size;
    char *tmp;
    struct cio_memfs *mf = ch->backend;

    if (count == 0) {
        return 0;
    }
```

**Comment by edsiper** [comment]
thanks guys for reporting and troubleshooting this.

Indeed there were two issues involved: a possible invalid non-null terminated Tag plus a wrong offset when writing/positioning data.

The fix is abf83a46a42826d7b175b9e247587ae130ff1841, any quick feedback is appreciated.

**Comment by edsiper** [comment]
Fixed in GIT master and 1.0 branch. v1.0.1 release have the fix.

thanks again.
